pkg/volume/emptydir/empty_dir.go

```diff
@@ -301,12 +301,17 @@ func (ed *emptyDir) assignQuota(dir string, mounterSize *resource.Quantity) erro
 		if err != nil {
 			klog.V(3).Infof("Unable to check for quota support on %s: %s", dir, err.Error())
 		} else if hasQuotas {
-			klog.V(4).Infof("emptydir trying to assign quota %v on %s", mounterSize, dir)
-			err := fsquota.AssignQuota(ed.mounter, dir, ed.pod.UID, mounterSize)
-			if err != nil {
-				klog.V(3).Infof("Set quota on %s failed %s", dir, err.Error())
+			_, err := fsquota.GetQuotaOnDir(ed.mounter, dir)
+			if err == nil {
```

Comment:
It feels like the logic in this function is backwards to usual golang practices. The functionality doesn't look wrong; just had to read it twice and has nesting that doesn't need to be here.

wdyt  (pretty sure I preserved the logic, but please double check):

```golang
_, err := fsquota.GetQuotaOnDir(ed.mounter, dir)
if err != nil {
  klog.V(4).Infof("Attempt to check quota on dir %s failed: %v", dir, err)
  // this is not a fatal error so return nil
  return nil
}
klog.V(4).Infof("emptydir trying to assign quota %v on %s", mounterSize, dir)
if err := fsquota.AssignQuota(ed.mounter, dir, ed.pod.UID, mounterSize); err != nil {
  klog.V(3).Infof("Set quota on %s failed %s", dir, err.Error())
  return err
}
return nil
```

Reply:
This can reduce one level of nesting. It looks better.

Updated.